packages/mds-event-processor/src/proc-event.ts

```diff
@@ -0,0 +1,271 @@
+import { dataHandler } from './proc'
+import db from '@mds-core/mds-db'
+import cache from '@mds-core/mds-cache'
+import stream from '@mds-core/mds-stream'
+
+import { getAnnotationData, getAnnotationVersion } from './annotation'
+import {
+  CE_TYPE,
+  StateEntry,
+  TripEvent,
+  TripTelemetry,
+  EVENT_STATUS_MAP,
+  VEHICLE_EVENT,
+  VEHICLE_REASON
+} from '@mds-core/mds-types'
+
+/*
+    Event processor that runs inside a Kubernetes pod.
+    Streams cloudevents from mds agency and process them in multiple ways:
+
+        1) quality checks
+        2) status changes
+        3) trip identification
+
+    Processed events/telemetry are added to various caches keyed as follows:
+
+        1) device:state (latest event/telemetry for a device)
+        2) trip:events (events linked to trips of a device)
+        3) device:ID:trips (all telemetry linked to all trips of a device)
+            - ID is the combination 'provider_id:device_id'
+
+    A Postgres table is also populated to store historical states:
+
+        REPORTS_DEVICE_STATES:
+          PRIMARY KEY = (provider_id, device_id, timestamp, type)
+          VALUES = deviceState
+*/
+
+async function eventHandler() {
+  await dataHandler('event', async function(type: CE_TYPE, data: any) {
+    console.log(type, data)
+    return processRaw(type, data)
+  })
+}
+
+async function processRaw(type: CE_TYPE, data: any) {
+  const { timestamp, device_id, provider_id, recorded } = data as {
+    timestamp: number
+    device_id: string
+    provider_id: string
+    recorded: number
+  }
+  const lastState = JSON.parse(await cache.hget('device:state', provider_id + ':' + device_id)) as StateEntry
+  // Construct state
+  const baseDeviceState = {
+    vehicle_type: 'scooter',
```

Comment:
Please import `VEHICLE_TYPES` from `mds-types` and replace this raw string with `VEHICLE_TYPES.scooter` instead.